res/controllers/common-hid-packet-parser.js

```diff
@@ -415,17 +413,19 @@ class HIDPacket {
             return;
         }
 
-        for (let byte_index = 0; byte_index < bytes; byte_index++) {
-            const index = field.offset + byte_index;
-            if (signed) {
-                if (value >= 0) {
-                    data[index] = (value >> (byte_index * 8)) & 255;
-                } else {
-                    data[index] = 255 - ((-(value + 1) >> (byte_index * 8)) & 255);
-                }
-            } else {
-                data[index] = (value >> (byte_index * 8)) & 255;
-            }
+        const dataView = new DataView(data.buffer);
+        if (field.pack === "b") {
+            dataView.setInt8(field.offset, value);
```

Comment:
fyi, not sure if you're aware, we might want to consider pulling in [structjs](https://github.com/lyngklip/structjs) into mixxx. In case we ever create a hid-packet-parser 2.0, it might allow a much nicer API for declaring fixed size layout packets.

Reply:
Interesting, but for parsing HID reports we need code, that can handle values of abritary length in bits, which are not byte aligned.

Reply:
Mhmm good point, that could be added in retrospect.